include/pagmo_plugins_nonfree/worhp.hpp

```diff
@@ -0,0 +1,1203 @@
+        }
+    }
+    /// Get numeric options.
+    /**
+     * @return the name-value map of optimisation numeric options.
+     */
+    std::map<std::string, double> get_numeric_options() const
+    {
+        return m_numeric_opts;
+    }
+    /// Set bool option.
+    /**
+     * This method will set the optimisation integer option \p name to \p value.
+     * The optimisation options are passed to the WORHP API when calling evolve().
+     *
+     * @param name of the option.
+     * @param value of the option.
+     */
+    void set_bool_option(const std::string &name, bool value)
+    {
+        m_bool_opts[name] = value;
+    }
+    /// Set bool options.
+    /**
+     * This method will set the optimisation integer options contained in \p m.
+     * It is equivalent to calling set_bool_option() passing all the name-value pairs in \p m
+     * as arguments.
+     *
+     * @param m the name-value map that will be used to set the options.
+     */
+    void set_bool_options(const std::map<std::string, bool> &m)
+    {
+        for (const auto &p : m) {
+            set_integer_option(p.first, p.second);
+        }
+    }
+    /// Get bool options.
+    /**
+     * @return the name-value map of optimisation integer options.
+     */
+    std::map<std::string, bool> get_bool_options() const
+    {
+        return m_bool_opts;
+    }
+    /// Clear all integer options.
+    void reset_integer_options()
+    {
+        m_integer_opts.clear();
+    }
+    /// Clear all numeric options.
+    void reset_numeric_options()
+    {
+        m_numeric_opts.clear();
+    }
+    /// Clear all numeric options.
+    void reset_bool_options()
+    {
+        m_bool_opts.clear();
+    }
+    /// Get the result of the last optimisation.
+    /**
+     * @return the result of the last call to WORHP. You can check
+     * The WORHP user manual for the meaning of the various entries.
+     * \verbatim embed:rst:leading-asterisk
+     * 
+     * .. seealso::
+     *
+     *    https://worhp.de/latest/download/user_manual.pdf
+     *
+     * \endverbatim
+     */
+    std::string get_last_opt_result() const
+    {
+        return m_last_opt_res;
+    }
+
+    /// Object serialization
+    /**
+     * This method will save/load \p this into the archive \p ar.
+     *
+     * @param ar target archive.
+     *
+     * @throws unspecified any exception thrown by the serialization of the UDA and of primitive types.
+     */
+    template <typename Archive>
+    void serialize(Archive &ar)
+    {
+        ar(cereal::base_class<not_population_based>(this), m_worhp_library, m_last_opt_res, m_integer_opts,
+           m_numeric_opts, m_bool_opts, m_screen_output, m_verbosity, m_f_cache, m_g_cache);
+    }
+
+private:
+    struct pair_hash {
+        template <class T1, class T2>
+        std::size_t operator()(const std::pair<T1, T2> &p) const
+        {
+            return std::hash<T1>()(p.first) ^ std::hash<T2>()(p.second);
```

Comment:
There's a boost ``hash_combine`` function that we should use for this sort of things, rather than rolling our own.

Reply:
changed to:

```
    struct pair_hash {
        template <class T1, class T2>
        std::size_t operator()(const std::pair<T1, T2> &p) const
        {
            std::size_t seed = 0;
            boost::hash_combine(seed, p.first);
            boost::hash_combine(seed, p.second);
            return seed;
        }
    }
```
 Not sure if now boost will be calling std::hash internally, but I guess it will use its own hash for each of the first, second